frontend/src/app/reports/reports.component.ts

```diff
@@ -0,0 +1,301 @@
+import {Component, OnInit} from '@angular/core';
+import {ReportsService} from './reports.service';
+import * as c3 from 'c3';
+import {ChartAPI} from 'c3';
+import {FormControl, Validators} from "@angular/forms";
+
+@Component({
+  selector: 'app-reports',
+  templateUrl: './reports.component.html',
+  styleUrls: ['./reports.component.css']
+})
+export class ReportsComponent implements OnInit {
```

Comment:
Whitespace in this class is a little awkward. Try running IntelliJ's autoformat on it.